### DISCUSSION TITLE
Storing cookies for multiple browser instances not working anymore after version 79

### DISCUSSION THREAD

**Discussion by KFucik** [cefsharp/CefSharp#3351] [Q&A]
Hi,

we are using CefSharp.Forms and .WPF in our application. Users are allowed to open multiple browser instances and each instance has its own RequestContext. Thus cookies should be stored per-browser. This worked fine with CefSharp v79. Starting with CefSharp v81 cookies are stilled written to disk. But on next application start all cookies are gone again.

That's how we initialize the browser for WPF:
```
var settings = new CefSettings
{
	LogFile = Path.Combine(basePath, "Cef.log"),
	LogSeverity = LogSeverity.Verbose
};
Cef.Initialize(settings, false, (IApp)null);

WebBrowser = new ChromiumWebBrowser
{
       RequestContext = new RequestContext(new RequestContextSettings
       {
             CachePath = Path.Combine(basePath, "CefSharp1")
       })
};
```
I uploaded a small repro project to https://drive.google.com/file/d/1RIoUsdYUaw1AUIsias9mOnP4TRhNDRhf/view?usp=sharing that will start two browsers. Each with its own RequestContext and CachePath.

Each and every time the application is started, Google will require to accept cookies. The selection is not stored.
When switching the same project back to CefSharp v79, the cookies will be correctly saved and Google only asks for acceptance of cookies a single time.

Is that a bug in CefSharp or are any code changes necessary to make it work in newer versions?

Best Regards,
Kevin

**Comment by amaitland** [comment]
> When using a RequestContext you must ensure that RequestContextSettings.CachePath is equal to or a child of CefSettings.RootCachePath See #3111 for details

As per https://github.com/cefsharp/CefSharp/releases/tag/v81.3.100 changes are required when using a RequestContext.  

Please read https://github.com/cefsharp/CefSharp/issues/3111#issuecomment-629713608

You need to specify CefSettings.RootCachePath and ensure all RequestContextSettings.CachePath paths are children of RootCachePath.

**Reply by KFucik** [reply]
Hi amaitland,

thank you very much for the hint. It's working perfectly fine again with RootCachePath!

Greetings
